python/graphstorm/eval/evaluator.py

```diff
@@ -1020,3 +1020,214 @@ def get_val_score_rank(self, val_score):
+        self._history = []
+        self.tracker = None
+        self._best_val_score = None
+        self._best_test_score = None
+        self._best_iter = None
+
+        self._task_evaluators = task_evaluators
+        assert len(self.task_evaluators) > 1, \
+            "There must be evaluators for different tasks." \
+            f"But get onely get {len(self.task_evaluators)}"
+
+        self._metric_list = {
+            task_id: evaluator.metric_list for task_id, evaluator in self.task_evaluators.items()
+        }
+
+        self._eval_frequency = eval_frequency
+        # TODO(xiang): Support early stop
+        assert use_early_stop is False, \
+            "GSgnnMultiTaskEvaluator do not support early stop now."
+        self._do_early_stop = use_early_stop
+
+        # add this list to store all of the performance rank of validation scores for pick top k
+        self._val_perf_rank_list = []
+
+
+    # pylint: disable=unused-argument
+    def do_early_stop(self, val_score):
+        """ Decide whether to stop the training
+
+        Note: do not support early stop for multi-task learning.
+        Will support it later.
+
+        Parameters
+        ----------
+        val_score: float
+            Evaluation score
+        """
+        raise RuntimeError("GSgnnMultiTaskEvaluator.do_early_stop is not implemented")
+
+    def get_metric_comparator(self):
+        """ Return the comparator of the major eval metric.
+
+            Note: not support now.
+
+        """
+        raise RuntimeError("GSgnnMultiTaskEvaluator.get_metric_comparator is not implemented")
+
+    # pylint: disable=unused-argument
+    def get_val_score_rank(self, val_score):
+        """
+        Get the rank of the given validation score by comparing its values to the existing value
+        list.
+
+        Note: not support now.
+
+        Parameters
+        ----------
+        val_score: dict
+            A dictionary whose key is the metric and the value is a score from evaluator's
+            validation computation.
+        """
+        raise RuntimeError("GSgnnMultiTaskEvaluator.get_val_score_rank is not implemented")
+
+    @property
+    def task_evaluators(self):
+        """ Task evaluators
+        """
+        return self._task_evaluators
+
+    @property
+    def best_val_score(self):
+        """ Best validation score
+        """
+        best_val_score = {
+            task_id: evaluator.best_val_score \
+                for task_id, evaluator in self.task_evaluators.items()
+        }
+        return best_val_score
+
+    @property
+    def best_test_score(self):
+        """ Best test score
+        """
+        best_test_score = {
+            task_id: evaluator.best_test_score \
+                for task_id, evaluator in self.task_evaluators.items()
+        }
+        return best_test_score
+
+    @property
+    def best_iter_num(self):
+        """ Best iteration number
+        """
+        best_iter = {
```

Comment:
Should this variable name be consistent with the calling method, i.e., `best_iter_num`?